src/vignette/storage/s3.clj

```diff
@@ -91,8 +91,9 @@
     (when-let [object (safe-get-object (add-timeouts :get (:creds this)) bucket path)]
       (when (valid-s3-get? object)
         (let [stream (:content object)
-              meta-data (:metadata object)]
-          (create-stored-object stream meta-data)))))
+              meta-data (:metadata object)
+              file-name (last (clojure.string/split (:key object) #"/"))]
```

Comment:
Could this be `nil`? What should the behavior be in that case? Should we use the hash?